onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -8,284 +8,224 @@
 
 import re
 import subprocess
 import json
 
-
-def augment_graph(model, quantization_candidates=['Conv', 'MatMul'], black_nodes=[], white_nodes=[]):
-    '''
-    Adds ReduceMin and ReduceMax nodes to all quantization_candidates op type nodes in
-    model and ensures their outputs are stored as part of the graph output
-        parameter model: loaded FP32 ONNX model to quantize
-        parameter quantization_candidates: node op types for nodes to be quantized.
-                                           Calibraton will be done for them.
-        parameter black_nodes: nodes with these names will be force ignored by this
-                               calibration augmentation, no mather what's their op type.
-        parameter white_nodes: nodes with these names will be force to be calibration augmented.
-        return: augmented ONNX model
-    '''
-
-    added_nodes = []
-    added_outputs = []
-    for node in model.graph.node:
-        should_be_calibrate = ((node.op_type in quantization_candidates) and
-                               (node.name not in black_nodes)) or (node.name in white_nodes)
-        if should_be_calibrate:
-            input_name = node.output[0]
-            # Adding ReduceMin nodes
-            reduce_min_name = ''
-            if node.name != '':
-                reduce_min_name = node.name + '_ReduceMin'
-            reduce_min_node = onnx.helper.make_node('ReduceMin', [input_name], [input_name + '_ReduceMin'],
-                                                    reduce_min_name,
-                                                    keepdims=0)
-            added_nodes.append(reduce_min_node)
-            added_outputs.append(helper.make_tensor_value_info(reduce_min_node.output[0], TensorProto.FLOAT, ()))
-
-            # Adding ReduceMax nodes
-            reduce_max_name = ''
-            if node.name != '':
-                reduce_max_name = node.name + '_ReduceMax'
-            reduce_max_node = onnx.helper.make_node('ReduceMax', [input_name], [input_name + '_ReduceMax'],
-                                                    reduce_max_name,
-                                                    keepdims=0)
-            added_nodes.append(reduce_max_node)
-            added_outputs.append(helper.make_tensor_value_info(reduce_max_node.output[0], TensorProto.FLOAT, ()))
-    model.graph.node.extend(added_nodes)
-    model.graph.output.extend(added_outputs)
-    return model
-
-
-# Using augmented outputs to generate inputs to quantize.py
-
-
-def get_intermediate_outputs(model_path, session, inputs, calib_mode='naive'):
-    '''
-    Gather intermediate model outputs after running inference
-        parameter model_path: path to augmented FP32 ONNX model
-        parameter inputs: list of loaded test inputs (or image matrices)
-        parameter calib_mode: type 'naive' gives (ReduceMin, ReduceMax) pairs
+import abc
+
+class CalibrationDataReaderInterface(metaclass=abc.ABCMeta):
```

Comment:
>CalibrationDataReaderInterface [](start = 6, length = 30)

IMO, it is better to name it without suffix Interface

Reply:
Okay, I see!